src/components/chips/js/chipsController.js

```diff
@@ -93,19 +129,38 @@
 
 
   /**
+   * Gets the index of an adjacent chip to select after deletion. Adjacency is
+   * determined as the next chip in the list, unless the target chip is the
+   * last in the list, then it is the chip immediately preceding the target. If
+   * there is only one item in the list, -1 is returned (select none).
+   * The number returned is the index to select AFTER the target has been
+   * removed.
+   * If the current chip is not selected, then -1 is returned to select none.
+   */
+  MdChipsCtrl.prototype.getAdjacentChipIndex = function(index) {
+    var len = this.items.length - 1;
+    if (this.selectChip != index || len == 0) {
+      return -1;
+    } else if (index == len) {
+      return index - 1;
+    } else {
+      return index;
+    }
+  };
+
+
+  /**
    * Append the contents of the buffer to the chip list. This method will first
    * call out to the md-chip-append method, if provided
+   * @param newChip
    */
-  MdChipsCtrl.prototype.appendChipBuffer = function() {
-    var newChip = this.getChipBuffer();
-    if (this.useMdChipAppend && this.mdChipAppend) {
+  MdChipsCtrl.prototype.appendChip = function(newChip) {
+    if (this.useMdChipAppend && !!this.mdChipAppend) {
```

Comment:
You never need `!!` in conditionals like `if`, `while`, etc. You only would use `!!` if you have a function that returns or is expecting a `boolean` and nothing else.